src/vdom/component.js

```diff
@@ -68,13 +68,14 @@ export function renderComponent(component, opts, mountAll, isChild) {
 		state = component.state,
 		context = component.context,
 		previousProps = component.prevProps || props,
-		previousState = component.prevState || state,
+		previousState = extend({}, component.prevState || state),
 		previousContext = component.prevContext || context,
 		isUpdate = component.base,
 		nextBase = component.nextBase,
 		initialBase = isUpdate || nextBase,
 		initialChildComponent = component._component,
 		skip = false,
+		snapshot = previousContext,
```

Comment:
Q: were we deviating from React here by passing `context` as a 3rd argument to `componentDidUpdate`?  If so, maybe we correct that here.

Reply:
Yep, we were/are deviating from React there. React only never had a third `context` parameter. Same for `shouldComponentUpdate` where we pass an additional context arg too. I'd prefer to these changes in a separate PR, because they are independent from `getSnapshotBeforeUpdate`.

Reply:
Agreed regarding the separate PR, but we should make sure they go out in the same release - we'll save a few bytes by dropping that reassignment.

Reply:
mh I have a branch where I removed the non-standard context arguments, but I get the exact same gzip size as in master :S